icepyx/core/read.py

```diff
@@ -363,13 +363,18 @@ def _add_var_to_ds(is2ds, ds, grp_path, wanted_groups_tiered, wanted_dict):
                 pass
 
             try:
-                # DevNote: these lines may cause a NumPy Warning, as explained here: https://numpy.org/doc/stable/release/1.11.0-notes.html?
-                # The below line will silence this warning...
-                # warnings.filterwarnings(
-                #     "default", category=DeprecationWarning, module=np.astype()
-                # )
-                is2ds["data_start_utc"] = is2ds.data_start_utc.astype(np.datetime64)
-                is2ds["data_end_utc"] = is2ds.data_end_utc.astype(np.datetime64)
+                if is2ds["data_end_utc"].data[0] == "Z":
```

Comment:
This fixed an AttributeError for me, but loading an ATL06 dataset still doesn't work. I'll do a bit more investigation first.

```suggestion
                if (
                    hasattr(is2ds, "data_end_utc")
                    and is2ds["data_end_utc"].data[0] == "Z"
                ):
```

Reply:
I pushed a modified version of this - it probably wasn't making it into the if statement because I forgot to compare just the last character...